pexpect/pxssh.py

```diff
@@ -143,6 +143,7 @@ def __init__ (self, timeout=30, maxread=2000, searchwindowsize=None,
         # used to set shell command-line prompt to UNIQUE_PROMPT.
         self.PROMPT_SET_SH = r"PS1='[PEXPECT]\$ '"
         self.PROMPT_SET_CSH = r"set prompt='[PEXPECT]\$ '"
+        self.PROMPT_SET_ZSH = r"PS1='[PEXPECT]%(!.#.$) '"
```

Comment:
This is sufficient for regular `zsh` but fails for users with prompt themes from `oh-my-zsh`, which overrides the `PS1` setting.

Reply:
Fixed in 686fa6f with `prompt restore`.